### PR TITLE
windows: add the sample ecs configurations into the images

### PR DESCRIPTION
**Pull request by rawahars**
## Summary
We are providing inbuilt configurations for ECS users for the following destinations-
- Amazon CloudWatch
    - New log groups would be created as `/aws/ecs/<CLUSTER_NAME>/<TASK_DEFINITION_FAMILY>`
    - New log streams would be created for each task in above log group with name as `task-out.<TASK_ID>.<CONTAINER_NAME>`
- Amazon S3
    - The logs would be uploaded in the provided bucket under key `<TASK_ID>/<CONTAINER_NAME>/%Y/%m/%d/%H/%M/%S`
- Amazon Kinesis Streams
    - All logs would be sent to the specified Kinesis Stream.
- Amazon Kinesis Firehose
    - All logs would be sent to the specified Data Stream.

We add the changes here so that-
- These configurations are baked into the Windows container images.
- Configured the entrypoint script so that it can be overridden with a custom config file location.

## Testing
Built new container images with these changes and then ran them by overriding the entrypoint with the following-
```
Powershell -Command C:\\entrypoint.ps1 -ConfigFile C:\\ecs_windows\\cloudwatch.conf
```
```
Powershell -Command C:\\entrypoint.ps1 -ConfigFile C:\\ecs_windows\\s3.conf
```
```
Powershell -Command C:\\entrypoint.ps1 -ConfigFile C:\\ecs_windows\\kinesis.conf
```
```
Powershell -Command C:\\entrypoint.ps1 -ConfigFile C:\\ecs_windows\\firehose.conf
```
## Description of changes:
windows: add the sample ecs configurations into the images

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Dockerfile.windows [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
 
 ADD certs /certs
 
-ADD AWS_FOR_FLUENT_BIT_VERSION /AWS_FOR_FLUENT_BIT_VERSION
-
 ADD entrypoint.ps1 /entrypoint.ps1
 
+# Add the version file.
+RUN Powershell -Command New-Item -Path C:/ -Name AWS_FOR_FLUENT_BIT_VERSION -ItemType "file" -Value $env:AWS_FOR_FLUENT_BIT_VERSION
+
```
Comment (PettitWesley) [inline Dockerfile.windows:26]: nit: I think this should be a separate commit

Comment (rawahars) [reply Dockerfile.windows:26]: rectified the same in the latest revision. The commits have been segregated.

File: ecs_windows/README.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
+## Using default Amazon ECS configurations for Windows images
```
Comment (PettitWesley) [inline ecs_windows/README.md:1]: Can we call the directory storing these configs `ecs_windows_forward_daemon` and then make it clear in title this is for running Fluent Bit as a daemon in ECS using forward plugin. 

And then as soon as the blog launches we link it here?

And then in the future, we can easily just have another dir: `ecs_windows_tail_daemon`

File: ecs_windows/README.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+In order to allow Amazon ECS Windows customers to get started promptly with Fluent Bit without performing significant configuration, we have crafted default configurations for the following output destinations-
+* [Amazon CloudWatch](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows/cloudwatch.conf)
+* [Amazon S3](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows/s3.conf)
+* [Amazon Kinesis Data Streams](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows/kinesis.conf)
+* [Amazon Kinesis Data Firehose](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows/firehose.conf)
+
+These default configurations would be baked in the Windows container images. In order to use the same out of the box with Amazon ECS, you need to ensure-
```
Comment (PettitWesley) [inline ecs_windows/README.md:9]: "would be" => are

Comment (PettitWesley) [reply ecs_windows/README.md:9]: We might want to put the version number of the first release with these

File: ecs_windows/README.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+* You have set the environment variables required for the specific configuration as documented later in this guide.
+* You have configured your Amazon ECS task definition for your application tasks to use fluentd logging driver and the following labels are configured in `labels` option of fluentd docker logging driver-
+
+    * com.amazonaws.ecs.cluster
+    * com.amazonaws.ecs.container-name
+    * com.amazonaws.ecs.task-arn
+    * com.amazonaws.ecs.task-definition-family
+    * com.amazonaws.ecs.task-definition-version
+
+These labels would be added by Amazon ECS agent to each container.
```
Comment (PettitWesley) [inline ecs_windows/README.md:21]: "are added by default" reads better IMO

File: ecs_windows/cloudwatch.conf [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+[OUTPUT]
+    Name                cloudwatch_logs
+    Match               out.*
+    region              ${AWS_REGION}
+    log_group_name      fallback-group
+    log_group_template  /aws/ecs/$ecs_cluster.$ecs_task_definition_family
+    log_stream_prefix   task-
+    auto_create_group   On
+    net.dns.resolver    LEGACY
```
Comment (PettitWesley) [inline ecs_windows/cloudwatch.conf:16]: it might be worth adding a note in the readme about this

File: ecs_windows/README.md (hunk 3 above)
Comment (rawahars) [reply ecs_windows/README.md]: Did this change.

File: ecs_windows/cloudwatch.conf (hunk 5 above)
Comment (rawahars) [reply ecs_windows/cloudwatch.conf]: Added a note about the default CloudWatch configuration.

In the other PR, I am planning to add a note about `net.dns.resolver    LEGACY` in the main README.

File: ecs_windows/README.md (hunk 2 above)
Comment (rawahars) [reply ecs_windows/README.md]: That's a good idea. I have made the changes here.

File: ecs_windows/README.md (hunk 4 above)
Comment (rawahars) [reply ecs_windows/README.md]: Did this change.

File: ecs_windows_forward_daemon/README.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
+# Default configurations for Windows images in Amazon ECS
+## Using Fluent Bit on Amazon ECS Windows in a Daemon configuration using forward input plugin
+
+In order to allow Amazon ECS Windows customers to get started promptly with Fluent Bit without performing significant configuration, we have crafted default configurations for the following output destinations-
+* [Amazon CloudWatch](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows_forward_daemon/cloudwatch.conf)
+* [Amazon S3](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows_forward_daemon/s3.conf)
+* [Amazon Kinesis Data Streams](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows_forward_daemon/kinesis.conf)
+* [Amazon Kinesis Data Firehose](https://github.com/aws/aws-for-fluent-bit/blob/mainline/ecs_windows_forward_daemon/firehose.conf)
+
+These default configurations are baked in the Windows container images with image versions 2.28.4 and above. These can be used in a Daemon configuration which uses forward input plugin.
```
Comment (PettitWesley) [inline ecs_windows_forward_daemon/README.md:10]: nit, I think this is slightly unclear: "in a Daemon Service deployment with a configuration that uses the forward.."

**Review by PettitWesley — APPROVED** [review]
Please see one new language nit comment

**Review by PettitWesley — APPROVED** [review]
